### ISSUE TITLE
Use google docstyle docs

### ISSUE THREAD

**Issue by david-caro** [inspirehep/inspire-schemas#107]
Add the sphinx plugin for it (see https://github.com/inveniosoftware-contrib/json-merger/tree/master/json_merger for an example)

Check if it can validate that the params defined in the docstring matches the params in the function, that would be great.

**Comment by kaplun** [comment]
For my understanding: what is the benefit of the Google docstring flavor? Why has it been chosen? How does it play WRT the parallel sprint made by upstream Invenio devs on finalizing documentation for all Invenio modules?

**Comment by david-caro** [comment]
Yep, I think we should formalize the way we make decisions somewhere (like storing the rationale or something).

Anyhow:

> what is the benefit of the Google docstring flavor?
Readability, that's the main advantage of it, it's way more readable for a human, ex:
```python
def something(one, two):
    """Does somethnig.

    :param one: first param
    :type one: str
    :param two: second param
    :type two: int
    :retrun: the answer to the most important question of life, the universe and everything.
    :rtype: int
    :rises: TowelNotFound when there's no towel to be found.
    :rises: UnrisableException when it can't rise an exception.
    :example:
        something('ni', 41)
    """

## vs

def something(one, two):
    """Does somethnig.

    Args:
        one (str): first param
        two (int): second param

    Returns:
        int: the answer to the most important question of life, the universe and everything.

    Rises:
        TowelNotFound when there's no towel to be found.
        UnrisableException: when it can't rise an exception.

    Example:
        something('ni', 41)
    """
```

And according to @jacquerie it will also complain when a parameter is missing in the docstring or when a parameter in the docstring is not in the function signature anymore (helping getting the docs in sync with the code).

> Why has it been chosen?
Because we did a technical meeting to decide about documentation processes and standards, and that's what we did :) (we can discuss it on another meeting if you want to discuss it further).

> How does it play WRT the parallel sprint made by upstream Invenio devs on finalizing documentation for all invenio modules?
It's totally independent on our side, we will still use as many docs from invenio as we can (to avoid duplicating work) but all the other aspects are only inspire-side.

**Comment by kaplun** [comment]
> > Why has it been chosen?

> Because we did a technical meeting to decide about documentation processes and standards, and that's what we did :) (we can discuss it on another meeting if you want to discuss it further).

Hehe, I know. Was more a redundant question on the pros and cons.

Can we use both restructured text and Google one at the same time?

**Comment by kaplun** [comment]
> And according to @jacquerie it will also complain when a parameter is missing in the docstring or when a parameter in the docstring is not in the function signature anymore (helping getting the docs in sync with the code).

This is super-cool!

**Comment by david-caro** [comment]
Not 100% sure, but I'd say at 75% that yes :)

**Comment by michamos** [comment]
> Can we use both restructured text and Google one at the same time?

Should we?

**Comment by david-caro** [comment]
I guess that he wants to know if we can mix them so instead of having to move all the docstrings at the same time, we can go migrating them step by step instead.

**Comment by david-caro** [comment]
Just checked, it will only complain if there's something documented that's not in the signature, with a bit cryptic message:

```
/home/dcaro/Work/david-caro/python-autosemver/autosemver/api.py:docstring of autosemver.api.get_changelog:None: WARNING: py:obj reference target not found: rpm
```

**Comment by jacquerie** [comment]
Closed in 507984eacf044951b43b3f4f493491e9b381624b. Note that GitHub does not understand "Closes #X and #Y", you have to write explicitly "Closes #X and closes #Y". See: https://help.github.com/articles/closing-issues-via-commit-messages/#closing-multiple-issues

CC: @ammirate

**Comment by michamos** [comment]
I wrote that commit message, good to know!
